src/lib/crypto/transmitters.ts

```diff
@@ -229,11 +229,11 @@ export abstract class Transmitter<Specs, Payload extends ITransactionPayload<Spe
                 .catch((reason: any) => { throw new TimeoutError(globalTimeout, receipt, reason) }),
             
             promisePoller({
-                taskFn: () => this.provider.getTransaction(hash),
+                taskFn: () => this.provider.getTransaction(hash).catch(err => err),
```

Comment:
What is the expected behavior here? if we use `.catch(e=>e)` we will always return a value instead of handling the error. [Example here](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABFAFgJzgdwKJo2xACgEpEBvAKEWsTQFMoQ0kw7NEAFDAWxgGc6hQvT5wANgDc6AGlp0AVnWikAvAD5yVGttQZ2rdrnyEA5AHk4ABz4niW6gF87Diq4CGfAJ6REoSLAREbjcYJBJNbQgEPig5PhAxWJVEN0wQ2N0sIzg0EgA6CDcoCBRCOkR1RDo7SOjxOjyxOABzUxEEqBNZdsSamiiwUTEGptaTbAAPJRAApAHYMBA6G2dEV2DQkiA)